src/content/blog/2024-07-05-whats-coming-next-for-eslint.md

```diff
@@ -0,0 +1,57 @@
+
+Here's what to expect going forward (with no specific timetable as everything is dependent upon contributor availability):
+
+* **`@eslint/js`** - We will slowly move all of the JavaScript-related functionality into the `@eslint/js` package, including rules and documentation. We will reuse the existing [Espree repo](https://github.com/eslint/espree/issues/609) and convert it into a monorepo containing Espree, `eslint-scope`, `eslint-visitor-keys`, and all of the core JavaScript rules. This will allow us to deal with everything JavaScript-related in a single repo.
+* **`@eslint/json`** - Our first language plugin will allow ESLint to lint JSON files natively. This plugin will contain the parsing logic and all of the related rules and documentation. Similar to `@eslint/json`, this will allow us to have a single repo to focus on JSON linting, further moving any language-specific functionality away from the core.
+* **`@eslint/markdown`** - [`eslint-plugin-markdown`](https://github.com/eslint/eslint-plugin-markdown) will be renamed to `@eslint/markdown` to better align with the other packages. We'll add the ability to lint Markdown and Markdown-specific rules.
+
+The current plan is to have these three official ESLint language plugins with the hopes that they serve as examples for the ecosystem to develop more language plugins.
+
+With language-specific functionality extract into separate repositories, that leaves us with the task of rewriting the core completely.
+
+## Core rewrite
+
+The architecture of the code in the ESLint repository hasn't changed considerably since ESLint was first created 11 years ago. As a result, there is a lot of technical debt that has accumulated and prevents us from making some of the changes we'd like to make. Some of the problems we currently face:
+
+* **Synchronous core logic.** The `Linter` class is completely synchronous, meaning we can't support asynchronous rules or asynchronous parsing, two feature requests we receive frequently. We'd need to either add a second class that works asynchronously or add alternate methods to `Linter` that worked asynchronously. Both options looked like significant work, effectively rewriting huge parts of the core.
+* **Limited API.** The public API is limited because ESLint, at the start, was never intended to be used as an API. The initial `linter` object was meant only to enable a browser demo. It was later replaced by the `Linter` class, and even later, an `ESLint` class to fully mimic the command line interface. When we want to expose new functionality, it needs to exist one of these two similar-sounding APIs, and unfortunately, the main decision point is whether or not the API needs to be available in the browser. If so, then it needs to be in `Linter`, otherwise it needs to be in `ESLint`. This is both confusing and not sustainable in the long term.
```

Comment:
```suggestion
* **Limited API.** The public API is limited because ESLint, at the start, was never intended to be used as an API. The initial `linter` object was meant only to enable a browser demo. It was later replaced by the `Linter` class, and even later, an `ESLint` class to fully mimic the command line interface. When we want to expose new functionality, it needs to exist in one of these two similar-sounding APIs, and unfortunately, the main decision point is whether or not the API needs to be available in the browser. If so, then it needs to be in `Linter`, otherwise it needs to be in `ESLint`. This is both confusing and not sustainable in the long term.
```